packages/core/docs/general/key-concepts.md

```diff
@@ -2,122 +2,61 @@
+## Progressive technology
 
-- `nuxt.config.js` to control your Nuxt App and frontend-related features of Vue Storefront;
-- `middleware.config.js` to add, configure and extend your integrations;
+It comes as no surprise that Vue Storefront is mostly based on Vue-related technologies: Vue.js itself,
+and Nuxt.js, which is built on top of it. While Vue.js provides tools to create user interfaces and 
+templates, Nuxt.js takes care of server-side rendering, routing, and internationalization.
 
-You can read more about it on the [Configuration](/guide/configuration.html) page.
-
-## Composables
-
-::: tip Composables? Is this a French meal?
-Composable is a function that uses [Vue.js Composition API](https://v3.vuejs.org/guide/composition-api-introduction.html) under the hood. It's commonly named with a `use` prefix (eg. `useProduct`, `useCart`). This convention comes from the React community which has a similar pattern called [Hooks](https://reactjs.org/docs/hooks-intro.html), which inspired Vue.js core team to introduce the Composition API. If you are not familiar with this concept, we strongly recommend checking the basics in the official  [Vue.js documentation](https://v3.vuejs.org/guide/composition-api-introduction.html).
-:::
-
-**Composables are the main public API of Vue Storefront** and in many cases the only API except configuration you'll work with.
-
-You can treat each composable as an independent micro-application. They manage their state, handle SSR and in most cases don't interact with each other (`useUser` and `useCart` are the exceptions).
-
-```js
-const { search, product, loading, error } = useProduct();
-```
-
-You can read more about them on the [Composables](/guide/composables.html) page.
-
-## Routing
-
-Out of the box, some routes are injected via `@vue-storefront/nuxt-theme` module:
-
-- Home Page (`/`);
-- Category Page (`/c/:slug_1/:slug_2?/:slug_3?/:slug_4?/:slug_5?`);
-- Product Page (`/p/:id/:slug/`);
-- User Profile Page (`/my-account/:pageName?`);
-- Checkout (`/checkout`):
-  - Shipping (`/checkout/shipping`);
-  - Billing (`/checkout/billing`);
-  - Payment (`/checkout/payment`);
-  - Thank You page (`/checkout/thank-you`);
-- Custom 404 page;
-
-Some [integrations](/integrations) may register additional routes. For example, CMS integrations often override Home Page and add custom, dynamic pages.
+It is recommended to get familiar with [Vue.js](https://vuejs.org/) and [Nuxt.js](https://nuxtjs.org/)
+before starting development in Vue Storefront.
 
-To override existing routes or adding your own, use [extendRoutes](https://nuxtjs.org/guides/configuration-glossary/configuration-router#extendroutes) in `nuxt.config.js`. Additionally, Nuxt.js automatically registers components created in the `pages` folder as new routes. You can read more about this on the [File System Routing](https://nuxtjs.org/docs/2.x/features/file-system-routing/) page.
+## Easy configuration
```

Comment:
Move this section below `Extendability and custom integrations` where we explain what `Server Middleware` is.